runtime/src/lib.rs

```diff
@@ -309,7 +311,21 @@ pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, Call, SignedExt
 /// Executive: handles dispatch to the various modules.
 pub type Executive = executive::Executive<Runtime, Block, system::ChainContext<Runtime>, Runtime, AllModules>;
 
+decl_runtime_apis! {
+	/// The API to query anchoring info.
+	pub trait AnchorApi {
+		fn get_anchor_by_id(id: Hash) -> Option<AnchorData<Hash, BlockNumber>>;
```

Comment:
I think there is a crucial piece missing here and that is for the caller to be able to find out until when the anchor is valid. Meaning the return would be better served by containing the expiration date as well.

Reply:
@pstehlik Technically speaking the there is already a storage function RPC autogenerated for that because of the `AnchorEvictDates get(get_anchor_evict_date): map T::Hash => u32;` index in storage. That's why we are omitting it here. Also if the anchor is invalid it would have been evicted and this function would not return any data, other than an error.